### PR TITLE
[feature](nereids)support stats estimation for is-null predicate

### PR DESCRIPTION
**Pull request by englefly**
## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by englefly** [comment]
run buildall

**Review by Kikyou1997 — APPROVED** [review]

File: fe/fe-core/src/main/java/org/apache/doris/nereids/stats/FilterEstimation.java [hunk 1]
```diff
@@ -386,6 +387,19 @@ public Statistics visitNot(Not not, EstimationContext context) {
...
     }
 
+    @Override
+    public Statistics visitIsNull(IsNull isNull, EstimationContext context) {
+        ColumnStatistic childStats = ExpressionEstimation.estimate(isNull.child(), context.statistics);
+        if (childStats.isUnKnown()) {
+            return new StatisticsBuilder(context.statistics).build();
+        }
+        double outputRowCount = context.statistics.getRowCount() - childStats.numNulls;
+        StatisticsBuilder builder = new StatisticsBuilder(context.statistics);
```
Comment (XieJiann) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/stats/FilterEstimation.java:397]: isNull output nulls, so the rowCount is childStast.numNulls?

Comment (englefly) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/stats/FilterEstimation.java:397]: 走偏了，想着 is not null 去了

**Comment by englefly** [comment]
run buildall

**Review by keanji-x — APPROVED** [review]

**Review by xzj7019 — APPROVED** [review]

**Review by morrySnow — DISMISSED** [review]

**Comment by englefly** [comment]
run buildall

**Comment by englefly** [comment]
run buildall

**Comment by englefly** [comment]
run buildall

**Comment by englefly** [comment]
run buildall

**Review by starocean999 — APPROVED** [review]

**Review by keanji-x — APPROVED** [review]

**Pull request merged**